test/common/runtime/runtime_impl_test.cc

```diff
@@ -623,27 +623,45 @@ TEST_F(StaticLoaderImplTest, RuntimeFromNonWorkerThreads) {
   // Now set up a test thread which verifies foo -> bar
   //
   // Then change foo and make sure the test thread picks up the change.
+  bool read_bar = false;
+  bool updated_eep = false;
   Thread::MutexBasicLockable mutex;
   Thread::CondVar foo_read;
   Thread::CondVar foo_changed;
   const Snapshot* original_thread_snapshot_pointer = nullptr;
   auto thread = Thread::threadFactoryForTest().createThread([&]() {
-    Thread::LockGuard lock(mutex);
-    EXPECT_EQ("bar", loader_->threadsafeSnapshot()->get("foo"));
-    original_thread_snapshot_pointer = loader_->threadsafeSnapshot().get();
-    EXPECT_EQ(original_thread_snapshot_pointer, loader_->threadsafeSnapshot().get());
-    foo_read.notifyOne();
-
-    foo_changed.wait(mutex);
-    EXPECT_EQ("eep", loader_->threadsafeSnapshot()->get("foo"));
+    {
+      Thread::LockGuard lock(mutex);
+      EXPECT_EQ("bar", loader_->threadsafeSnapshot()->get("foo"));
+      read_bar = true;
+      original_thread_snapshot_pointer = loader_->threadsafeSnapshot().get();
+      EXPECT_EQ(original_thread_snapshot_pointer, loader_->threadsafeSnapshot().get());
+      foo_read.notifyOne();
+    }
+
+    {
+      Thread::LockGuard lock(mutex);
+      while (!updated_eep) {
+        foo_changed.wait(mutex);
+      }
+      EXPECT_EQ("eep", loader_->threadsafeSnapshot()->get("foo"));
+    }
   });
 
   {
```

Comment:
I think we can remove this scope, now that we have two more for the lock

Reply:
yep.